core/shared/src/test/scala/fs2/concurrent/ChannelSuite.scala

```diff
@@ -278,9 +278,17 @@ class ChannelSuite extends Fs2Suite {
   }
 
   test("sendAndClose closes right after sending the last element in synchronous case") {
-    val channel = Channel.synchronous[IO, Int]
+    val test = Channel
+      .synchronous[IO, Int]
+      .flatMap { ch =>
+        ch.stream
+          .concurrently(Stream.emit(0).covary[IO].evalMap(ch.sendAndClose))
+          .compile
+          .drain *> ch.isClosed
+      }
+      .assertEquals(true)
 
-    checkIfSendAndCloseClosing(channel).parReplicateA_(if (isJVM) 10000 else 1)
+    test.parReplicateA_(if (isJVM) 10000 else 1)
```

Comment:
Actually this particular test I don't think we need to `replicateA`: there isn't a race condition because the test has one possible outcome. This is unlike the other test where there is a race, and depending on the winner there are multiple outcomes that we have to check for.

Reply:
Oh okay, I will refactor it. Thanks!